dsl/src/commonTest/kotlin/com/freeletics/flowredux/dsl/StateMachine.kt

```diff
@@ -1,24 +1,46 @@
 package com.freeletics.flowredux.dsl
 
-import kotlin.coroutines.EmptyCoroutineContext
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.DelicateCoroutinesApi
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.consumeAsFlow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.launch
 
 @OptIn(ExperimentalCoroutinesApi::class, FlowPreview::class)
 fun StateMachine(
         builderBlock: FlowReduxStoreBuilder<TestState, TestAction>.() -> Unit
-): FlowReduxStateMachine<TestState, TestAction> {
-    val sm = object : FlowReduxStateMachine<TestState, TestAction>(
-        TestState.Initial,
-        CoroutineScope(Dispatchers.Default),
-        CommandLineLogger
-    ){
-        init {
-            spec(builderBlock)
+): TestStateMachine {
+    return TestStateMachine(builderBlock)
+}
+@OptIn(ExperimentalCoroutinesApi::class, FlowPreview::class)
+class TestStateMachine  constructor(
```

Comment:
It's a pity that we cannot use `FlowReduxStateMachine` ... I am afraid that in the long run we could encounter issues specific to `FlowReduxStateMachine` that are not tested in `TestStateMachine`.


With that said, what we most want to test is the DSL, so it should ok if we could then write some sort of other Unit test that would cover the `FlowReduxStateMachine` internal behavior. At the moment I have a hard time to imagine how we could test `FlowReduxStateMachine`, do you see how we could do this? I still end up writing an integration test that would cover `FlowReduxStateMachine` plus DSL ...

Reply:
Testing `FlowReduxStateMachine` should be generally fine as long as the test doesn't do something like 
```
onEnter {
  OverrideState(OtherState)
}
```